### ISSUE TITLE
Problematic ASGD Optimizer

### ISSUE THREAD

**Issue by tiancaishaonvjituizi** [pytorch/pytorch#74884]
### 🐛 Describe the bug

I saw an "Optimizers documentation tracking issue" https://github.com/pytorch/pytorch/issues/63236. But ASGD is overlooked. Sadly, there is at least three problems in ASGD Optimizer:

1. The learning rate of ASGD Optimizer cannot be scheduled by lr_scheduler
1. In [document](https://pytorch.org/docs/stable/generated/torch.optim.ASGD.html), "eta" is indeed the "current learning rate".
2. Argument "lambd" and "weight_decay" are almost all the same: "lambd" itself is already the coefficient of L2 regularization. The only difference is that "lambd" is used to update the eta (a.k.a. learning rate) in the ASGD Optimizer built-in lr scheduling strategy. However, ASGD, as an optimizer, should not be coupled with any given lr scheduling strategy. Once the learning rate can be controled by lr scheduler, "lambd" and "weight_decay" will be completely all the same. (See https://github.com/npinto/bottou-sgd/blob/master/README.txt for the origin of "lambd" and "eta")

### Versions

master
